lib/modules/commentNavigator.js

```diff
@@ -198,6 +199,10 @@ let currentCategory = '';
 function onScroll() {
 	const category = currentCategory;
 	if (category) {
+		// Non-linear comment categories are not sorted by distance from top,
+		// so the most matching comment cannot determined by the algorithm below
+		if (sortTypes[category].nonlinear) return;
```

Comment:
Would `if (currently selected post is on screen) return;` also work? I think I'd prefer that so that there's still _some_ degree of functionality for `popular`